server/app/views/admin/programs/ProgramBlocksView.java

```diff
@@ -1080,10 +1136,19 @@ private boolean hasNoRepeatedBlocks(ProgramDefinition programDefinition, long bl
     return programDefinition.getBlockDefinitionsForEnumerator(blockId).isEmpty();
   }
 
+  /**
+   * Returns if the currently displayed program is a draft or not. This is used to identify if the
+   * view should be editable or not.
+   */
   private boolean viewAllowsEditingProgram() {
     return programDisplayType.equals(DRAFT);
   }
 
+  /**
+   * Creates the Edit button shown at the top of the page. For a read only view it starts changes to
+   * an edit view. For an editable view it leads to the page that allows editing the Program
```

Comment:
Maybe "For a read only view, it redirects to an editable view."

Reply:
Done